test/routes/home_route_test.dart

```diff
@@ -67,4 +71,43 @@ void main() {
       expect(ContentControl.instance.state.allSongs.songs, [songWith()]);
     });
   });
+
+  testWidgets('app shows exit confirmation toast if enabled in the preferences', (WidgetTester tester) async {
+    await Prefs.confirmExitingWithBackButton.set(true);
+    await tester.runAppTest(() async {
+      final SystemChannelObserver systemObserver = SystemChannelObserver(tester);
+      final ToastChannelObserver toastObserver = ToastChannelObserver(tester);
+      await BackButtonInterceptor.popRoute();
+      expect(toastObserver.lastToastMessage, l10n.pressOnceAgainToExit,
+          reason: 'Expected the app to ask for confirmation before exiting');
+      expect(systemObserver.closeRequests, 0,
+          reason: 'Expected the app not to close after showing the toast');
```

Comment:
While it might seem good have a human readable sentence to the failure, to me it clutters the test a bit, since it's faster to just look at the expect to understand what went wrong

Can we get rid of the `reason:` everywhere, what do you think? If you disagree, it's ok if we keep them :-)

Reply:
I will examine where they actually add any value.

Reply:
Done, see cf334940c306eb4eb5e88a90d79efc4fa31be798.